docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1260 @@
+![Sequence Diagram for Add Task Command](diagrams/class/jpeg/add_task.jpg)
+
+<hr>
+TaskList also allows the deletion of tasks by the user.
+
+The figure below shows how the delete task command works:
+![Sequence Diagram for Delete Task Command](diagrams/class/jpeg/SequenceDiagram_RemoveTimetable.jpg)
+
+Here are the general steps that the command goes through when the user inputs "delete 1":
+1. The `CliUserInterface` receives the "delete 1" input by the user and passes it to the `InputParser` class.
+2. `InputParser` parses the input to determine the type of command and the index of the task that is required to delete.
+The Parser then constructs a `RemoveCommandTimetableMode` with constructor as shown below.
+```java
+public RemoveCommandTimetableMode(int indexToRemove, AppState uiMode) {
+    this.setAppState(uiMode);
+    this.indexToRemove = indexToRemove;
+}
+```
+3. Method **execute()** then calls the `TaskList` stored in `AppState` to update the deletion of the task.
+It also constructs `CliMessages` to display messages to the user.
+4. If the deletion is successful, `CliMessages` displays the message to the user.
+
+##### 4.2.1.2. Design Considerations
+
+###### Aspect: How to store tasks in `TaskList`
+- **Alternative 1 (current choice):** Store as an `ArrayList` of tasks
+    - Pros: It is easier to implement because the code base are list based.
+    - Cons: It is unoptimized in terms of complexity, which requires more work for scaling of the application.
+- **Alternative 2:** Store as a Hash Table with the key as the index and value as `Task`
+    - Pros: It has a better time complexity and reduce the work in scaling stage since this data structure is more optimized (O(1) can be achieved).
+    - Cons: It takes more resources to implement. The constant factor for a hashing algorithm is significant and not
+     worth the tradeoff for smaller amounts of data, like a typical user would be likely to generate.
+
+#### 4.2.2. Mark as done feature  
+
+##### 4.2.2.1. Implementation  
+
+The `Task` class contains a member `isDone` of Boolean type.
+
+The following sequence diagram shows how the mark as done operation works:  
+
+// TODO Insert diagram inside 
+
+The following is an example of the processes that occur when the user uses the mark as done function:  
+
+Step 1. The user types `done 1`. The `done 1` command is passed through `InputParser#getCommandFromInput`, which constructs a DoneCommandTimetableMode object and calls `DoneCommandTimetable#execute()`.  
+
+Step 2. `execute()` is called, which then initialises a variable `taskList` of type TaskList. The method then calls `AppState#getTaskList`, which returns all exisiting tasks in the current tasklist.  
+
+Step 3. The `execute()` method proceeds to parse the user's intended task index to be marked as done through `Integer.parseInt()`. Using this index, it initialises a varible `taskDone` of type task and calls `AppState#markAsDone(index)`.  
```

Comment:
Spelling mistake with 'variable'. Isnt the variable already initialized to False and this method changes it to True?